### PR TITLE
Move estimateGasAndNonce from wallet to txMgr, and add gasLimit 1.1x bump by default

### PR DESCRIPTION
**Pull request by samlaf**
See https://github.com/Layr-Labs/eigensdk-go/pull/165#issuecomment-2025966763

### CODE REVIEW

**Review by shrimalmadhur — DISMISSED** [review]

**Comment by samlaf** [comment]
@ian-shim no rush but I'm still going to wait for your review given you have more context on this.
Also @shrimalmadhur was reviewing on his phone so I trust his review 50%. :D

File: chainio/txmgr/txmgr.go [hunk 1]
```diff
@@ -120,3 +150,56 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txID wallet.TxID) *t
...
+		m.log.Info("eth_maxPriorityFeePerGas is unsupported by current backend, using fallback gasTipCap")
+		gasTipCap = FallbackGasTipCap
+	}
+
+	header, err := m.client.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	gasFeeCap := new(big.Int).Add(header.BaseFee, gasTipCap)
```
Comment (ian-shim) [inline chainio/txmgr/txmgr.go:171]: setting `gasFeeCap` to `2 * baseFee + gasTipCap` has pretty cool advantages if you want to use it: https://www.blocknative.com/blog/eip-1559-fees

**Review by ian-shim — APPROVED** [review]
lol how is @shrimalmadhur reviewing PRs?

File: chainio/txmgr/txmgr.go [hunk 2]
```diff
@@ -15,6 +16,11 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+var (
+	FallbackGasTipCap          = big.NewInt(15_000_000_000)
```
Comment (gpsanant) [inline chainio/txmgr/txmgr.go:20]: super high imo

Comment (gpsanant) [reply chainio/txmgr/txmgr.go:20]: 5gwei should be plenty

Comment (samlaf) [reply chainio/txmgr/txmgr.go]: [b733a74](https://github.com/Layr-Labs/eigensdk-go/pull/167/commits/b733a74ae6525e3ddbc5eaee2aa73815e900e037)

File: chainio/txmgr/txmgr.go (hunk 1 above)
Comment (samlaf) [reply chainio/txmgr/txmgr.go]: [90da821](https://github.com/Layr-Labs/eigensdk-go/pull/167/commits/90da821787ea16ea4b4f4f9343a6420213baa389)

**Comment by samlaf** [comment]
@ian-shim increase gasLimit to 1.20x following our discussions: [d81fb38](https://github.com/Layr-Labs/eigensdk-go/pull/167/commits/d81fb38966cb5e5541d92b507859bc6ee1793c86)

**Review by ian-shim — APPROVED** [review]

**Pull request merged**